### PR TITLE
Handle nodes that aren't leafy.

### PR DESCRIPTION
**Pull request by calebcase**
This relates to:

https://github.com/akesterson/dpath-python/issues/119

And incorporates some of the changes from:

https://github.com/akesterson/dpath-python/pull/120
https://github.com/akesterson/dpath-python/pull/122
https://github.com/akesterson/dpath-python/pull/123

### CODE REVIEW

File: dpath/segments.py
Comment (dargueta) [inline dpath/segments.py]: 1) Typo: this should be `False` not `false`. The fact that the tests pass mean this path isn't getting hit.
2) I recommend catching `TypeError` instead of a bare `except:`, as a bare `except` will also mask exceptions thrown by the interpreter due to syntax errors, block keyboard interrupts, etc.

Comment (calebcase) [reply dpath/segments.py]: Good catch, the linter apparently needs adjusted. It was configured to catch these at one point.

**Comment by wsantos** [comment]
@calebcase can you push the fix ? what do we need to get this merged ? I've merged this to my fork if you guys want to use until this get merged, https://github.com/wsantos/dpath-python

**Comment by dargueta** [comment]
This PR is still broken though; see my comments above

**Comment by wsantos** [comment]
So what is the plan to get this done @dargueta @calebcase ?

**Comment by dargueta** [comment]
@calebcase hasn't fixed the bug I found so we can't do anything with it until they're done.

**Comment by moomoohk** [comment]
Hi. I can't really use this library without a fix for this issue. If this PR is abandoned should I open another one? Will it get reviewed and merged?
Any other alternatives in the meanwhile?

**Comment by dargueta** [comment]
> If this PR is abandoned should I open another one?

There have been four already; really it's up to @calebcase to fix the bug in this PR and @akesterson to merge it. @akesterson you may be able to modify this PR yourself since you're the owner.

**Comment by moomoohk** [comment]
Looking forward to seeing this fix merged then. In the meantime I'm using @wsantos's fork

**Comment by akesterson** [comment]
Blocked by #136

**Comment by calebcase** [comment]
@moomoohk I'm going to abandon this, but it might be worth incorporating it in another PR.

**Comment by moomoohk** [comment]
Hi everyone. I'm reopening this PR for review and merging.

**Review by bigsablept — APPROVED** [review]
This looks good to me. Thanks for getting it ready!

**Comment by moomoohk** [comment]
Changes will be pushed to PyPI soon enough

**Pull request merged**
